### ISSUE TITLE
Wrong typescript definition for "when" method

### ISSUE THREAD

**Issue by hasfoug** [ecrmnn/collect.js#317]
Hi there,

I've noticed that the typescript definition for "when" method is wrong

`when(condition: boolean, fn: (this: any) => any, defaultFn: (this: any) => any): void;`

In reality, the first callback function is most likely gonna return the modified collection, so I assume the return type of the method should be the Collection itself.

The current definition makes the compiler think that chained methods after the "when" method call are called on void, while in reality, it is not true because they are can be called on the modified collection. 

E.g.

`
collect(items).when(condition1, (collection) => collection.push(newItem)).when(condition2, (collection) => collection.pop())
`

I suppose that void also makes sense sometimes though, so it would be good if the return type is 

`Collection<Item> | void`

Also, the second callback defaultFn should be optional, right now it is required in the typescript definition which also makes the compiler throw warnings. 

To sum up, the right definition in my opinion should be next:

`when(condition: boolean, fn: (this: any) => any, defaultFn?: (this: any) => any): Collection<Item> | void;`

**Comment by ecrmnn** [comment]
Hi, @hasfoug.

I'm not seasoned enough in TypeScript to comment on this.
If this is the correct definition I'd be happy to merge a pull request.

**Comment by hasfoug** [comment]
@ecrmnn 

Thanks for your reply, I've created the PR with the changes https://github.com/ecrmnn/collect.js/pull/318

**Comment by hasfoug** [comment]
Hi @ecrmnn,

Could you merge it?
